state-chain/pallets/cf-witnesser/src/lib.rs

```diff
@@ -461,7 +467,7 @@ pub mod pallet {
 				(last_expired_epoch..=current_epoch)
 					.all(|epoch| CallHashExecuted::<T>::get(epoch, call_hash).is_none())
 			{
-				if let Some(mut extra_data) = ExtraCallData::<T>::get(epoch_index, call_hash) {
+				if let Some(mut extra_data) = ExtraCallData::<T>::take(epoch_index, call_hash) {
```

Comment:
We might as well delete the clean up code now then? And just write a migration to clean up anything that's currently there?

Reply:
It will clean itself up over the course of a couple of epochs, and finding the values that can be deleted now will require iterating over all the votes, checking the vote count etc. I'd rather save the time it would take develop and test this.

Reply:
ok, let's create an issue at least so we don't forget about

Reply:
What I mean is that it's ok to forget about it - the values get pruned after 2 epochs anyway.